.github/workflows/uml.yml

```diff
@@ -0,0 +1,37 @@
+name: Update the UML Diagrams
+
+on:
+  pull_request:
+    types: [closed]
+    branches: [main]
```

Comment:
This would only be on merge to main branch. With that, we won't have to have it run all the time. i.e. all the pngs updating every PR

WDYT?

Reply:
Set it up that it doesn't push if there wasn't a change. So "every PR" is likely not true. 

Does knowing this change when this action should be run @juanitorduz ?